addons/project/views/project_views.xml

```diff
@@ -1521,7 +1524,7 @@
                     <field name="legend_done" invisible="1"/>
                     <field name="legend_blocked" invisible="1"/>
                     <field name="kanban_state" widget="state_selection" optional="show" options="{'hide_label': True}" nolabel="1" required="0"/>
-                    <field name="stage_id" invisible="context.get('set_visible',False)" optional="show" readonly="not context.get('default_project_id')"/>
+                    <field name="stage_display" invisible="context.get('set_visible',False)" optional="show" readonly="not context.get('default_project_id')"/>
```

Comment:
This is good when there are multiple projects in the context, but if we are viewing the tasks of a single project, we want to be able to edit this value.
To achieve this, I think we will need to create a view inheriting this one, where the `stage_id` field will be replaced by `stage_display`, like this: 
```xml
<record id="new_view_inheriting_the_list_view" model="ir.ui.view">
    <field name="name">Name</field>
    <field name="model">project.task</field>
    <field name="inherit_id" ref="view_task_tree2"/>
    <field name="mode">primary</field>
    <field name="arch" type="xml">
        <field name="stage_id" position="replace">
            <field name="stage_display" optional="show" readonly="1"/>
        </field>
    </field>
</record>
```
We can then use this view in the `action_view_all_task` action by setting it here:
```diff
diff --git a/addons/project/views/project_views.xml b/addons/project/views/project_views.xml
index 8d3100ced0b..6a942fb4b70 100644
--- a/addons/project/views/project_views.xml
+++ b/addons/project/views/project_views.xml
@@ -1681,6 +1681,7 @@
         <record id="open_view_all_task_list_tree" model="ir.actions.act_window.view">
             <field name="sequence" eval="20"/>
             <field name="view_mode">tree</field>
+            <field name="view_id" ref="new_view_inheriting_the_list_view"/>
             <field name="act_window_id" ref="action_view_all_task"/>
         </record>
         <record id="open_view_all_task_list_calendar" model="ir.actions.act_window.view">
```
Let me know if you need help for this :slightly_smiling_face:

Reply:
I didn't know you could replace fields using the xml, i thought it was only possible using xpath on kanban.
Thanks a lot for the tips, it seems to be working,not 100% if it's okay :) tell me if anything is wrong!